### PR TITLE
Bom 1154

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Ticket:** [BOM-1154](https://openedx.atlassian.net/browse/BOM-1154)
### Description
- Enabled Django 2.0, 2.1 and 2.2 environments in tox.
- Added Travis tests for Django 1.11 and Django 2.0.
- Updated Jasmine constraint from 2.99.0 --> 3.1.0 as we dropped support for python2.0 and versions>3.1.0 cause test failures.

### CODE REVIEW

File: integrated_channels/integrated_channel/management/commands/transmit_learner_data.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ class Command(IntegratedChannelCommandMixin, BaseCommand):
     help = _('''
     Transmit Enterprise learner course completion data for the given EnterpriseCustomer.
     ''')
+    stealth_options = ('enterprise_customer_slug', 'user1', 'user2')
```
Comment (jmbowman) [inline integrated_channels/integrated_channel/management/commands/transmit_learner_data.py]: Why was this needed?

File: requirements/js_test.in [hunk 2]
```diff
@@ -1,6 +1,4 @@
 # Requirements for Jasmine javascript testing framework
 
-jasmine==2.9.0
-cherrypy==17.3.0   # This dependency shouldn't be needed.
-                   # It should just be a dependency of jasmine, but without
-                   # this pinning it upgraded which broke jasmine tests.
+jasmine==3.1.0  # versions>3.1.0 cause test failures.
+cherrypy
```
Comment (jmbowman) [inline requirements/js_test.in]: Given the comment above, can `cherrypy` be removed from this file now?

File: .travis.yml [hunk 3]
```diff
@@ -37,6 +40,11 @@ matrix:
         - RUNJSHINT=true
     - python: 3.6
       env: TOXENV=pii-annotations
+  allow_failures:
```
Comment (jmbowman) [inline .travis.yml]: These seem to be passing now, can this section be removed?  It's a problem if tests fail in the configuration that we're currently using in production.

Comment (UsamaSadiq) [reply .travis.yml]: Sure, I'll remove this section.

File: integrated_channels/integrated_channel/management/commands/transmit_learner_data.py (hunk 1 above)
Comment (UsamaSadiq) [reply integrated_channels/integrated_channel/management/commands/transmit_learner_data.py]: Before **Django2.0**, any arguments passed to the `call_command()` function are only evaluated on time of their usage in the function but after **Django2.0**, the arguments are validated during the function call. We can either add the arguments by `argparser` or can add in `stealth_options` as I did in my change. See [Django2.0 release_notes](https://docs.djangoproject.com/en/2.0/releases/2.0/#call-command-validates-the-options-it-receives) for the details.

File: requirements/js_test.in (hunk 2 above)
Comment (UsamaSadiq) [reply requirements/js_test.in]: I'll check it once on local by removing and if possible will remove it.

File: tox.ini [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = {py35,py36}-master
+envlist = py{35,36}-django111, py{35,36}-django{20,21,22}
```
Comment (mraarif) [inline tox.ini]: why not simply do `py{35,36}-django{111,20,21,22}`

Comment (UsamaSadiq) [reply tox.ini]: I was facing some issues in the format to decide so I followed the format of some other repo in edx. Changed the `envlist` as per suggestion.

File: requirements/edx-platform-constraints.txt [hunk 5]
```diff
@@ -41,18 +41,18 @@ django-babel-underscore==0.5.2
...
+django-config-models==2.0.0
 django-cors-headers==2.5.3
 django-countries==5.5
 django-crum==0.7.5
 django-fernet-fields==0.6
 django-filter==2.2.0
 django-ipware==2.1.0
 django-js-asset==1.2.2    # via django-mptt
-django-method-override==0.2.0
+git+https://gitlab.com/Ayub-khan/django-method-override.git@5270af321be2e576d8e8b3c4191711a19975c356#egg=django-method-override==1.0.4
```
Comment (Ayub-Khan) [inline requirements/edx-platform-constraints.txt]: is this coming from the platform ?

File: .travis.yml [hunk 6]
```diff
@@ -7,7 +7,10 @@ python:
   - 3.6
 
 env:
-  - TOXENV=master
+  - TOXENV=django111
+  - TOXENV=django20
+  - TOXENV=django21
+  - TOXENV=django22
```
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply .travis.yml]: Updated as per suggestion.

File: requirements/edx-platform-constraints.txt (hunk 5 above)
Comment (UsamaSadiq) [reply requirements/edx-platform-constraints.txt]: Yes, this file is generated automatically by running `make upgrade` and it fetches requirement constraints from edx-platform.
